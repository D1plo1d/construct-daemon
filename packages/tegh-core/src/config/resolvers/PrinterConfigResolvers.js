const PrinterConfigResolvers = {
  PrinterConfig: {
    components: (source, args) => {
      if (args.componentID != null) {
        const components = source.components.find(c => (
          c.id === args.componentID
        ))
        if (components == null) {
          return []
        }
        return [components]
      }
      return source.components
    },
    plugins: (source, args) => {
      if (args.package != null) {
        const plugin = source.plugins.find(p => p.package === args.package)
        if (plugin == null) {
          return []
        }
        return [plugin]
      }
      return source.plugins
    },
  },
}

export default PrinterConfigResolvers
