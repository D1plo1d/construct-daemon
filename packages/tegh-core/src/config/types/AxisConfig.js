import { Record } from 'immutable'
import uuid from 'uuid/v4'

export const AxisConfigFactory = Record({
  id: null,
  name: null,
  feedrate: null,
})

const AxisConfig = props => (
  AxisConfigFactory({
    id: props.id || uuid(),
    ...props(),
  })
)

export default AxisConfig
