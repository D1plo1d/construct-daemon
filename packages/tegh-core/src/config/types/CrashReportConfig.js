import { Record, Map } from 'immutable'
import uuid from 'uuid'

export const CrashReportConfigFactory = Record({
  id: null,
  modelVersion: 0,
  directory: '/var/log/tegh',
  uploadCrashReportsToDevs: true,
  extendedConfig: Map(),
})

const CrashReportConfig = ({
  id = uuid.v4(),
  modelVersion = 0,
  ...props
} = {}) => (
  CrashReportConfigFactory({
    ...props,
    id,
    modelVersion,
    extendedConfig: Map(props.extendedConfig),
  })
)

export default CrashReportConfig
