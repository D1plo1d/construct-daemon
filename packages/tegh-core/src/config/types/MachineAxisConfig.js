import { Record, Map } from 'immutable'
import t from 'tcomb-validation'

export const MachineAxisConfigStruct = t.struct({
  feedrate: t.Number,
})

const MachineAxisConfig = Record(
  Map(MachineAxisConfigStruct.meta.props).map(() => null).toJS(),
)

export default MachineAxisConfig
