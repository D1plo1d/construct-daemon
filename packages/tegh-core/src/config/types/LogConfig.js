import { Record, List, Map } from 'immutable'
import uuid from 'uuid/v4'

import { WARNING, ERROR, FATAL } from '../../log/types/logLevelEnum'

export const LogConfigFactory = Record({
  maxLength: 1000,
  stderr: List([WARNING, ERROR, FATAL]),
  extendedConfig: Map(),
})

const LogConfig = ({
  id = uuid(),
  ...props
} = {}) => (
  LogConfigFactory({
    ...props,
    id,
    extendedConfig: Map(props.extendedConfig),
  })
)

export default LogConfig
