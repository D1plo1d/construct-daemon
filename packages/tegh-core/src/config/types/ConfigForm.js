import { Record, Map } from 'immutable'
import t from 'tcomb-validation'

import LogConfig, { LogConfigStruct } from './LogConfig'
import CrashReportConfig, { CrashReportConfigStruct } from './CrashReportConfig'
import PrintFromLocalPathConfig, { PrintFromLocalPathConfigStruct } from './PrintFromLocalPathConfig'
import PluginConfig, { PluginConfigStruct } from './PluginConfig'
import MachineConfig, { MachineConfigStruct } from './MachineConfig'
import MaterialConfig, { MaterialConfigStruct } from './MaterialConfig'

export const configFormStructFields = {
  id: t.String,
  name: t.String,
  macros: t.dict(t.String, t.list(t.String)),
  log: LogConfigStruct,
  crashReports: CrashReportConfigStruct,
  printFromLocalPath: PrintFromLocalPathConfigStruct,
  materials: t.dict(t.String, MaterialConfigStruct),
  machine: MachineConfigStruct,
  plugins: t.dict(t.String, PluginConfigStruct),
}

export const ConfigFormStruct = t.struct(configFormStructFields)

export const ConfigFormRecordFactory = Record(
  Map(ConfigFormStruct.meta.props).map(() => null),
)

const mapOfRecords = (entries, recordFactory) => (
  Map(entries).map(props => recordFactory(props))
)

const ConfigForm = props => ConfigFormRecordFactory({
  ...props,
  log: LogConfig(props.log),
  crashReports: CrashReportConfig(props.crashReports),
  printFromLocalPath: PrintFromLocalPathConfig(props.printFromLocalPath),
  plugins: mapOfRecords(props.plugins, PluginConfig),
  machine: MachineConfig(props.machine),
  materials: mapOfRecords(props.materials, MaterialConfig),
})

export default ConfigForm
