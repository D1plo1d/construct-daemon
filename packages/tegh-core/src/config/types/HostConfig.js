import { Record, List, Map } from 'immutable'
import uuid from 'uuid/v4'

import CrashReportConfig from './CrashReportConfig'
import MaterialConfig from './MaterialConfig'
import LogConfig from './LogConfig'

export const HostConfigRecordFactory = Record({
  id: null,
  modelVersion: 0,
  name: null,
  crashReports: CrashReportConfig(),
  materials: List(),
  extendedConfig: Map(),
  log: LogConfig(),
})

const HostConfig = ({
  id = uuid(),
  modelVersion = 0,
  crashReports = {},
  materials = [],
  log = {},
  ...props
} = {}) => (
  HostConfigRecordFactory({
    ...props,
    id,
    modelVersion,
    crashReports: CrashReportConfig(crashReports),
    materials: materials.map(MaterialConfig),
    log: LogConfig(log),
    extendedConfig: Map(props.extendedConfig),
  })
)

export default HostConfig
