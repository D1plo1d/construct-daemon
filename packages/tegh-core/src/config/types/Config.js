import { Record, List } from 'immutable'
import uuid from 'uuid/v4'

import HostConfig from './HostConfig'
import PrinterConfig from './PrinterConfig'
import MaterialConfig from './MaterialConfig'
import developmentConfig from '../../../../../development.config'

export const ConfigRecordFactory = Record({
  id: null,
  modelVersion: 0,
  host: null,
  printer: null,
  materials: List(),
})

const Config = ({
  id = uuid(),
  modelVersion = 0,
  host = {},
  printer = {},
  materials = [],
  ...props
} = {}) => (
  ConfigRecordFactory({
    id,
    modelVersion,
    host: HostConfig(host),
    printer: PrinterConfig(printer),
    materials: List(materials).map(material => MaterialConfig(material)),
    ...props,
  })
)

export const MockConfig = ({
  host = {},
  printer = {},
  ...props
} = {}) => (
  Config({
    ...developmentConfig,
    host: {
      ...developmentConfig.host,
      ...host,
    },
    printer: {
      ...developmentConfig.printer,
      ...printer,
    },
    ...props,
  })
)

export default Config
