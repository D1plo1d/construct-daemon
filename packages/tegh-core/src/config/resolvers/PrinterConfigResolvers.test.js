import { Record } from 'immutable'
import typeDefs from 'tegh-schema'

import deterministicTestSetup from '../../util/testing/deterministicTestSetup'
import snapshotTestResolvers from '../../util/testing/snapshotTestResolvers'

import { MockConfig } from '../types/Config'

import PrinterConfigResolvers from './PrinterConfigResolvers'

describe('PrinterConfigResolvers', () => {
  it('presents a consistent API', () => {
    deterministicTestSetup()

    const state = Record({
      config: MockConfig(),
    })()

    snapshotTestResolvers({
      typeDefs,
      typeName: 'PrinterConfig',
      resolvers: PrinterConfigResolvers,
      rootValue: state.config.printer,
      contextValue: {
        store: {
          getState: () => state,
        },
      },
    })
  })
})
