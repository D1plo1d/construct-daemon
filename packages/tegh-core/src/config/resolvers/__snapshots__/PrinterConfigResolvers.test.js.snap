// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrinterConfigResolvers presents a consistent API 1`] = `
Object {
  "_testPrinterConfig": Object {
    "components": Array [
      Object {
        "id": "bbbserialController",
      },
      Object {
        "id": "x111",
      },
      Object {
        "id": "y111",
      },
      Object {
        "id": "z111",
      },
      Object {
        "id": "heater_1",
      },
      Object {
        "id": "abvwee1",
      },
      Object {
        "id": "ndrgrwef",
      },
      Object {
        "id": "ndrgrwef2",
      },
      Object {
        "id": "bdfbxb",
      },
    ],
    "fixedListComponentTypes": Array [
      "TOOLHEAD",
    ],
    "id": "aaxcvxcvcxv-bvb-csdf234231",
    "plugins": Array [
      Object {
        "id": "asgbvas23",
      },
      Object {
        "id": "aaabbbccc333",
      },
      Object {
        "id": "aaabbbccc123",
      },
    ],
  },
}
`;
